plugins/UsersManager/API.php

```diff
@@ -418,14 +478,95 @@ public function getSitesAccessFromUser($userLogin)
                     'access' => 'admin'
                 );
-
             }
             return $return;
         }
-
         return $this->model->getSitesAccessFromUser($userLogin);
     }
 
+    /**
+     * For each website ID, returns the access level of the given $userLogin (if the user is not a superuser).
+     * If the user doesn't have any access to a website ('noaccess'),
+     * this website will not be in the returned array.
+     * If the user doesn't have any access, the returned array will be an empty array.
+     *
+     * @param string $userLogin User that has to be valid
+     *
+     * @param int|null $limit
+     * @param int|null $offset
+     * @param string|null $filter_search text to search for in site name, URLs, or group.
+     * @param string|null $filter_access access level to select for, can be 'some', 'view' or 'admin' (by default 'some')
+     * @return array    The returned array has the format
+     *                    array(
+     *                        ['idsite' => 1, 'site_name' => 'the site', 'access' => 'admin'],
+     *                        ['idsite' => 2, 'site_name' => 'the other site', 'access' => 'view'],
+     *                        ...
+     *                    )
+     * @throws Exception
+     */
+    public function getSitesAccessForUser($userLogin, $limit = null, $offset = 0, $filter_search = null, $filter_access = null)
+    {
+        Piwik::checkUserHasSomeAdminAccess();
+        $this->checkUserExists($userLogin);
+
+        if (Piwik::hasTheUserSuperUserAccess($userLogin)) {
+            throw new \Exception("This method should not be used with superusers.");
+        }
+
+        $idSites = null;
+        if (!Piwik::hasUserSuperUserAccess()) {
+            $idSites = $this->access->getSitesIdWithAdminAccess();
+        }
+
+        list($sites, $totalResults) = $this->model->getSitesAccessFromUserWithFilters($userLogin, $limit, $offset, $filter_search, $filter_access, $idSites);
+        foreach ($sites as &$siteAccess) {
+            list($siteAccess['role'], $siteAccess['capabilities']) = $this->getRoleAndCapabilitiesFromAccess($siteAccess['access']);
+            $siteAccess['role'] = empty($siteAccess['role']) ? 'noaccess' : reset($siteAccess['role']);
+            unset($siteAccess['access']);
+        }
+
+        $hasAccessToAny = $this->model->getSiteAccessCount($userLogin) > 0;
+
+        return [
+            'results' => $sites,
+            'total' => $totalResults,
+            'has_access_to_any' => $hasAccessToAny
+        ];
+    }
+
+    /**
+     * Sets access of all sites matching the given filters. The access levels that are changed are for sites the user already has
+     * access to.
+     *
+     * @param string $userLogin The user whose access will be set.
+     * @param string $access The access to set.
+     * @param string|null $filter_search The
+     * @param string|null $filter_search text to search for in site name, URLs, or group.
```

Comment:
`filter_search` is mentioned twice and I wonder if the name could be more clear that it searches within websites?